### PR TITLE
Add requests to course page

### LINKED ISSUE #116: Missing the ability to accept or decline requests to join a course
**Requirements :**

- Add the ability for the course owner to accept or decline requests from students to join a course

### PR DESCRIPTION
**Pull request by RanyWh**
**This PR's goal is to give the course owner the ability to accept or decline requests from students to join the course.
The requests are shown inside the course page in the join course tab that's only visible for the teacher.**

Steps:

- Add student_course.html
- Add student_course.css
- Include student_course.html in course page
- Add in views and urls AcceptOrDeclineStudentCourse class
- Added tests for the accept and decline options

Course Page join course tab :
צילום מסך 2023-06-04 011651

dependencies  #114 
fixes #116

### CODE REVIEW

File: course/templates/course/course_page.html [hunk 1]
```diff
@@ -75,34 +57,37 @@
 
               <br></br>
               <label class="label">Course Description :</label>
-              <br></br>
-              <p>{{course.description}}</p>
+<<<<<<< HEAD
+              <p style="display:inline" >{{course.description}}</p>
+=======
+              <p style="display:inline">{{course.description}}</p>
+>>>>>>> ead046d (Ui improvment)
```
Comment (shirazyomtov) [inline course/templates/course/course_page.html:65]: There are leftover from the confict in git, Please remove this

File: course/templates/course/course_page.html [hunk 2]
```diff
@@ -75,34 +57,37 @@
...
-              Messages content
-              {% block Messages %} {% endblock %}
+<<<<<<< HEAD
+              {% block Messages %} 
+              {% include "student_course.html" %}
+              {% endblock %}
+=======
+            student content
+>>>>>>> ead046d (Ui improvment)
             </div>
```
Comment (shirazyomtov) [inline course/templates/course/course_page.html:89]: same as above

File: course/templates/course/courses_table.html [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+                    {% if user.profile.account_type == 'T'%}
+                    <td>{{course.get_pending_requset_count_for_course}}</td>
+                    {% endif %}
+                </tbody>
+                {% endfor %}
+            </table>
+        </div>
+        {% if user.profile.account_type == 'T'%}
+        <a href="add">
+            <!-- <button type="button" class="btn btn-success" style="float: right;">Add Course</button></a> -->
```
Comment (shirazyomtov) [inline course/templates/course/courses_table.html:45]: Please remove this comment

File: course/templates/course/courses_two_tables.html [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+                <td><a href="/course/{{course.course_id}}">{{course.course_name.capitalize}}</td>
+                <td>{{course.price}} ₪</td>
+                <td>{{course.get_category_display.capitalize}}</td>
+                <td>{{course.get_pending_requset_count_for_course}}</td>
+            </tbody>
+            {% endfor %}
+        </table>
+        <a href="add">
+            <!-- <button type="button" class="btn btn-success" style="float: right;">Add Course</button></a> -->
+            <button class="icon-btn add-btn">
```
Comment (shirazyomtov) [inline course/templates/course/courses_two_tables.html:34]: Please remove this comment

File: static/styles/courses.table.css [hunk 5]
```diff
@@ -4,5 +4,116 @@ height: 50px;
...
+  left: 15px;
+  height: 4px;
+  top: calc(50% - 2px);
+}
+
+.add-btn:hover .add-icon::after {
+  right: 15px;
+  height: 4px;
+  top: calc(50% - 2px);
+}
```
Comment (shirazyomtov) [inline static/styles/courses.table.css:119]: Missing a new line at the end of the file

File: static/styles/student_course.css [hunk 6]
```diff
@@ -0,0 +1,117 @@
...
+  left: 0.96em;
+  top: 0.7em;
+  width: 0.5em;
+  height: 0.75em;
+  border: solid white;
+  border-width: 0.2em 0.2em 0.2em 0.2em;
+  box-shadow: 0.1em 0.1em 0em 0 rgba(0,0,0,0.3);
+  transform: rotate(45deg);
+} */
+>>>>>>> ead046d (Ui improvment)
```
Comment (shirazyomtov) [inline static/styles/student_course.css:51]: There are leftover from the confict in git, Please remove this

File: static/styles/student_course.css [hunk 7]
```diff
@@ -0,0 +1,117 @@
...
+  border-radius: .5rem;
+  background-color: #0004;
+  visibility: hidden;
+}
+
+.scroll:hover::-webkit-scrollbar-thumb{ 
+  visibility: visible;
+}
+=======
+>>>>>>> ead046d (Ui improvment)
```
Comment (shirazyomtov) [inline static/styles/student_course.css:117]: There are leftover from the confict in git, Please remove this

File: tests/course/test_course_pages.py [hunk 8]
```diff
@@ -20,33 +23,61 @@ def teacher_course_zero(persist_user):
...
+
+@pytest.fixture
+def authorized_client_teacher(client, persist_profile_teacher):
+    client.force_login(persist_profile_teacher.user)
+    return client
+
+
+@pytest.fixture
+def authorized_client_teacher_and_student(client, persist_profile_teacher_and_student):
+    client.force_login(persist_profile_teacher_and_student.user)
```
Comment (shirazyomtov) [inline tests/course/test_course_pages.py:49]: Same comment as in this [PR](https://github.com/redhat-beyond/HighTeach/pull/114#discussion_r1216229754)

**Review by shirazyomtov — CHANGES_REQUESTED** [review]
Great work, left a few comments

**Review by Davidco94 — DISMISSED** [review]
LGTM

**Review by shirazyomtov — DISMISSED** [review]
LGTM

File: course/views.py [hunk 9]
```diff
@@ -9,10 +9,18 @@
 
 class CourseList(LoginRequiredMixin, View):
     def get(self, request):
-        courses = TeacherCourse.objects.get_teacher_courses(request.user)
-        courses |= TeacherCourse.objects.get_student_approved_teacher_courses(request.user)
-        context = {'courses': courses}
-        return render(request, 'course/courses.html', context)
+        if request.user.profile.account_type == 'B':
```
Comment (Omeramsc) [inline course/views.py:12]: A lot of conditional... where is the test that verify it works as expected?

File: tests/course/test_course_pages.py [hunk 10]
```diff
@@ -103,3 +105,30 @@ def test_show_course_page(self, authorized_client, persist_course):
...
+
+    def test_teacher_rejects_request_to_join_course(self, authorized_client_teacher, student_course0):
+        student_course_id = student_course0.student_course_id
+        teacher_course_id = student_course0.teacher_course_id.course_id
+        assert student_course0.status.value == 'Pending'
+        course_creation_response = authorized_client_teacher.post("/course/" + str(teacher_course_id) + '/' +
+                                                                  str(student_course_id),
+                                                                  {'Delete': 'Delete'}, follow=True)
+        assert course_creation_response.status_code == 200
+        assert not StudentCourse.objects.filter(student_course_id=student_course_id).exists()
```
Comment (Omeramsc) [inline tests/course/test_course_pages.py:131]: what about the student_course0.statue.value? does the request itself exists? is it "Rejected"?

**Review by Omeramsc — CHANGES_REQUESTED** [review]

File: tests/course/test_course_pages.py (hunk 10 above)
Comment (Omeramsc) [reply tests/course/test_course_pages.py:150]: Sounds like a bug. how the student can know about his request status? I'd argue it should be changed to "Rejected"...

Comment (Omeramsc) [reply tests/course/test_course_pages.py]: Very good! I left one comment regarding the test, fix that and it is ready IMO

**Review by Omeramsc — DISMISSED** [review]
LGTM, Good job

**Review by shirazyomtov — DISMISSED** [review]
LGTM

**Review by natifridman — COMMENTED** [review]
Looks good, please resolve conflicts

**Review by shirazyomtov — APPROVED** [review]
LGTM

**Review by Omeramsc — APPROVED** [review]

**Pull request merged**